src/convertToObject.js

```diff
@@ -6,7 +6,24 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const str = sourceString
+    .replace(/^\s*[\r\n]/gm, '')
+    .replace(/\s*;\s*/g, ';')
+    .replace(/\s*:\s*/g, ': ')
+    .trim();
+
+  const cleanArray = str.split(';').filter((elem) => elem.length > 0);
+  const obj = {};
+
+  for (let i = 0; i < cleanArray.length; i++) {
+    const properties = cleanArray[i].split(':');
+    const key = properties[0].trim();
+    const value = properties[1].trim();
+
+    obj[key] = value;
```

Comment:
Currently, all values are stored as strings in the resulting object. If the task requires converting numeric strings to numbers or handling other data types, additional parsing logic should be added here.